Sources/OpenCombine/Publishers/Publishers.Zip.swift

```diff
@@ -0,0 +1,765 @@
+//
+//  Publishers.Zip.swift
+//
+//  Created by Eric Patey on 29.08.2019.
+//
+
+#if canImport(COpenCombineHelpers)
+import COpenCombineHelpers
+#endif
+
+extension Publishers {
+
+    /// A publisher created by applying the zip function to two upstream publishers.
+    public struct Zip<UpstreamA: Publisher, UpstreamB: Publisher>: Publisher
+        where UpstreamA.Failure == UpstreamB.Failure
+    {
+
+        /// The kind of values published by this publisher.
+        public typealias Output = (UpstreamA.Output, UpstreamB.Output)
+
+        /// The kind of errors this publisher might publish.
+        ///
+        /// Use `Never` if this `Publisher` does not publish errors.
+        public typealias Failure = UpstreamA.Failure
+
+        public let a: UpstreamA
+
+        public let b: UpstreamB
+
+        public init(_ a: UpstreamA, _ b: UpstreamB) {
+            self.a = a
+            self.b = b
+        }
+
+        /// This function is called to attach the specified `Subscriber` to this
+        ///  `Publisher` by `subscribe(_:)`
+        ///
+        /// - SeeAlso: `subscribe(_:)`
+        /// - Parameters:
+        ///     - subscriber: The subscriber to attach to this `Publisher`.
+        ///                   once attached it can begin to receive values.
+        public func receive<Downstream: Subscriber>(subscriber: Downstream) where
+            UpstreamB.Failure == Downstream.Failure,
+            Downstream.Input == (UpstreamA.Output, UpstreamB.Output)
+        {
+            _ = Inner<Downstream>(downstream: subscriber, a, b)
+        }
+    }
+
+    /// A publisher created by applying the zip function to three upstream publishers.
+    public struct Zip3<
```

Comment:
@broadwaylamb there's no way to satisfy both this rule and your requirement for protocol conformance constraints to be within angle brackets instead of `where`. Which one would you choose then?

Reply:
...and we also need to fit these within 90 characters to avoid triggering the `line_length` warning.